src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+  const groupKeys: string[] = [];
+
+  Object.values(items).forEach((itemKey) => {
```

Comment:
Try to make it with 1 for each
For know you have a lot of action & hard alghoritm